### PR TITLE
MINIFI-37: Create a volatile repository and config items for

### PR DESCRIPTION
**Pull request by phrocker**
NoOp and Volatile repository configuration

Thank you for submitting a contribution to Apache NiFi - MiNiFi C++.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced
     in the commit message?

- [ ] Does your PR title start with MINIFI-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [ ] Has your PR been rebased against the latest commit within the target branch (typically master)?

- [ ] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the LICENSE file?
- [ ] If applicable, have you updated the NOTICE file?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -299,7 +299,19 @@ Additionally, users can utilize the MiNiFi Toolkit Converter (version 0.0.1 - sc
...
+### Configuring Volatile and NO-OP Repositories
+
+     in minifi.properties 
+     
+     # For Volatile Repositories:
+     nifi.flow.repository.class.name=VolatileRepository
+     nifi.provenance.repository.class.name=VolatileRepository
+     
+     # For NO-OP Repositories:
+	 nifi.flow.repository.class.name=NoOpRepository
```
Comment (apiri) [inline README.md]: I think this works for how flowfile is currently implemented but if we have a model closer to that of the Java counterparts, not sure this makes a lot of sense outside of provenance.  For the other repositories we would not capture the current state of where a flowfile is in its overall processing nor have any content available between processor hops in the case of content.  Let me know your thoughts.

File: libminifi/src/core/RepositoryFactory.cpp [hunk 2]
```diff
@@ -48,9 +50,15 @@ std::shared_ptr<core::Repository> createRepository(
   try {
     std::shared_ptr<core::Repository> return_obj = nullptr;
     if (class_name_lc == "flowfilerepository") {
+      std::cout << "creating flow" << std::endl;
```
Comment (phrocker) [inline libminifi/src/core/RepositoryFactory.cpp]: sorry I added that to be captured by a test, but I guess I forgot the outputstreamappender can redirect to stdout. I'll fix. Thanks for reminding me.

File: libminifi/src/core/RepositoryFactory.cpp [hunk 3]
```diff
@@ -48,9 +50,15 @@ std::shared_ptr<core::Repository> createRepository(
...
       return_obj = instantiate<core::repository::FlowFileRepository>();
     } else if (class_name_lc == "provenancerepository") {
       return_obj = instantiate<provenance::ProvenanceRepository>();
+    } else if (class_name_lc == "volatilerepository") {
+      return_obj = instantiate<repository::VolatileRepository>();
+    } else if (class_name_lc == "nooprepository") {
+      std::cout << "creating noop" << std::endl;
```
Comment (phrocker) [inline libminifi/src/core/RepositoryFactory.cpp]: it's needed for a test, but the test should be using the outputstreamappender. will fix. thanks.

File: libminifi/include/core/Repository.h [hunk 4]
```diff
@@ -47,14 +47,21 @@ namespace nifi {
 namespace minifi {
 namespace core {
 
+#define REPOSITORY_DIRECTORY "./repo"
```
Comment (phrocker) [inline libminifi/include/core/Repository.h]: Yes. agreed.

File: README.md (hunk 1 above)
Comment (phrocker) [inline README.md]: My thought, initially was that we could have processors that could be self contained and work with or without connections. In those cases the configuration wouldn't need to be updated but we may be able to disable both repositories at runtime thus altering certain devices to have a different capability. I support what you are suggesting though but wonder if the Java model fits all embedded devices and use cases.

Comment (apiri) [inline README.md]: That sounds fair.  Happy to leave it be for the time being and we can adjust as needed later.

File: libminifi/include/core/Repository.h (hunk 4 above)
Comment (apiri) [inline libminifi/include/core/Repository.h]: Created https://issues.apache.org/jira/browse/MINIFI-321

**Comment by phrocker** [comment]
@apiri after addressing your comments, above, I ran tests which made me rethink part of how this was tested. To cover my bases I'm going to test a little more and comment when it's ready again.

**Comment by apiri** [comment]
@phrocker Thanks for the heads up.  Will look for that update.

**Comment by phrocker** [comment]
@apiri This should be good once travis deems it so. I'm adding some information on the wiki on configuring this and other portions of MiNiFi cpp

**Comment by apiri** [comment]
reviewing updates

**Comment by apiri** [comment]
Looks good.  Will get this merged and thanks for your efforts.

**Pull request closed without merging**